packages/framework/src/Model/Product/Search/ElasticsearchSearchClient.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace Shopsys\FrameworkBundle\Model\Product\Search;
+
+use Elasticsearch\Client;
+
+class ElasticsearchSearchClient implements SearchClient
+{
+    /**
+     * @var \Elasticsearch\Client
+     */
+    private $client;
+
+    public function __construct(Client $client)
+    {
+        $this->client = $client;
+    }
+
+    /**
+     * @param int $domainId
+     * @param string|null $searchText
+     * @return int[]
+     */
+    public function search(int $domainId, $searchText): array
+    {
+        if (!$searchText) {
+            return [];
+        }
+
+        $parameters = $this->createQuery($domainId, $searchText);
+        $result = $this->client->search($parameters);
+        return $this->extractIds($result);
+    }
+
+    /**
+     * @param int $domainId
+     * @param string $searchText
+     * @return array
+     */
+    protected function createQuery(int $domainId, string $searchText): array
+    {
+        return [
+            'index' => $domainId,
+            'type' => '_doc',
+            'size' => 1000,
```

Comment:
Yes! And it can cause troubles that will be consulted during BV. Elasticsearch itself can return at most 10000 results, but it has horrible performance in postre later. This is a trade-off between accurance and number of resutls. And White team uses the same value.

The issue with number of products will be solved when the whole filter is done by elasticsearch.

Reply:
I see.